### PR TITLE
Update config-reference-systemxml.md

### PR DESCRIPTION
**Pull request by dfelton**
## Purpose of this pull request

- Added additional available validation classes.
- "Description" column had a mixture of actual descriptions, and often times a copy/paste of the validation failure message. Revised to more consistently be a description of the rule rather than list out the validation failure message.
  - Personal opinion is that a description of rule is more appropriate for a "developer" oriented guide, although also see value in providing the validation failure message as part of guide.
- For some rules, provide the regex that the value must match against.
- For some rules, provide the validation failure message.
- Clearly indicated that "required-entry" and "validate-no-empty" have the same validation test, but that each has a different validation failure message.
- Spacing on source code adjusted for readability of source (no impact on rendering).

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/config-guide/prod/config-reference-systemxml.html

## Links to Magento source code

-  https://github.com/magento/magento2/blob/2.4/app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js

## Misc

I think the additional validation rules are of high value. However, I have also begun changing some existing conventions of the current document in the process. I would like to hear from the community before I invest further time and for this reason am opening the PR now to gather that feedback. Will either revert certain changes or continue down the path I have started depending on feedback received.

### CODE REVIEW

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md [hunk 1]
```diff
@@ -359,28 +359,37 @@ The following validation rules are available:
...
+| `time12h`                       | Allows a valid time in 12-hour format, between 12:00 am and 11:59:59 pm. For example `3 am`, `11:30 pm`, `02:15:00 pm`. |
+| `mobileUK`                      | Allows a (UK) mobile phone number.                                                                                      |
+| `phoneUK`                       | Allows a (UK) phone number.                                                                                             |
+| `phoneUS`                       | Allows a (US) phone number.                                                                                             |
+| `required-entry`                | Disallows an empty value (equivalent validation as `validate-no-empty`).<br>Validation failure message: "This is a required field." |
+| `validate-no-html-tags`         | Disallows usage of HTML tags.                                                                                           |
+| `validate-select`               | Enforces a that the select option chosen not have a `null` value, string value of `none`, or string length of 0.        |
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:376]:
```suggestion
| `validate-select`               | Enforces that the select option chosen not have a `null` value, string value of `none` or string length of 0.        |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md [hunk 2]
```diff
@@ -359,28 +359,37 @@ The following validation rules are available:
...
+| `validate-emailSender`          | Allows a valid email address. For example, johndoe@domain.com.                                                          |
+| `validate-password`             | Allows 6 or more characters. Leading and trailing spaces will be ignored.                                               |
+| `validate-admin-password`       | Allows 7 or more characters, using both numeric and alphabetic.                                                         |
+| `validate-url`                  | Allows a valid URL. Protocol is required (http://, https:// or ftp://).                                                 |
+| `validate-clean-url`            | Allows a valid URL. For example, http://www.example.com or www.example.com.                                             |
+| `validate-xml-identifier`       | Allows a valid XML-identifier. For example, something_1, block5, id-4.                                                  |
+| `validate-ssn`                  | Allows a valid social security number. For example, 123-45-6789.                                                        |
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:391]:
```suggestion
| `validate-ssn`                  | Allows a valid (US) social security number. For example, 123-45-6789.                                                        |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md [hunk 3]
```diff
@@ -359,28 +359,37 @@ The following validation rules are available:
...
+| `validate-no-empty`             | Disallows an empty value (equivalent validation as `requried-entry`).<br>Validation failure message: "Empty value."     |
+| `validate-alphanum-with-spaces` | Allows usage of letters (a-z or A-Z), numbers (0-9), or spaces only.                                                    |
+| `validate-data`                 | Allows usage of letters (a-z or A-Z), numbers (0-9), or underscores (\_) only.<br>The first character must be a letter.<br>(Must match expression: `/^[A-Za-z]+[A-Za-z0-9_]+$/`)<br>Validation failure message: "Please use only letters (a-z or A-Z), numbers (0-9) or underscore (\_) in this field, and the first character should be a letter."  |
+| `validate-street`               | Allows usage of letters (a-z or A-Z), numbers (0-9), spaces, and “#” only.                                              |
+| `validate-phoneStrict`          | Allows a valid phone number. For example, (123) 456-7890 or 123-456-7890.                                               |
+| `validate-phoneLax`             | Allows a valid phone number. For example, (123) 456-7890 or 123-456-7890.                                               |
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:382]: What is the difference between Strict & Lax?

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Please also add these changes and additions to the 2.3 version of the page

Great extension to this page!

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md (hunk 3 above)
Comment (dfelton) [reply src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:382]: Good question. I remember these existing in M1 as well. off the top of my head I don't know but I can get back to you on this.

**Comment by dfelton** [comment]
> Please also add these changes and additions to the 2.3 version of the page

@BarnyShergold certainly. Will do as soon as I get some extra time this week.

**Comment by dobooth** [comment]
Lax: https://regex101.com/r/QLtvb3/3/
Strict: https://regex101.com/r/bNjZNi/2/
Lax allows single digit country codes.

**Comment by BarnyShergold** [comment]
@dfelton - Will you be completion work on this? We've not had an update for nearly 3 weeks

**Comment by keharper** [comment]
@dfelton @BarnyShergold This PR is at the threshold of being closed due to inactivity. I don't want to lose this information, so if I can get technical approval, I'll merge the PR and backport the info to 2.3 on the side.

**Comment by BarnyShergold** [comment]
Sounds a good move - I was going to pick this up with you tomorrow as it would be 2 weeks waiting for a response !

**Comment by BarnyShergold** [comment]
@keharper - Any news on getting this merged? If you do I can do a PR for 2.3?

**Comment by dobooth** [comment]
Happy new year. Can we move on this one?

**Review by keharper — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**